packages/cactus-plugin-ccmodel-hephaestus/src/main/typescript/plugin-ccmodel-hephaestus.ts

```diff
@@ -0,0 +1,911 @@
+          throughput: Number(
+            latency != 0
+              ? ((
+                  1 /
+                  ((latency + previousLatency) / numberOfCurrentEvents)
+                ).toFixed(3) as unknown as number)
+              : 0,
+          ),
+          latestUpdate: lastAggregated,
+        };
+        this.crossChainModel.setCCTxs(ccTxID, updatedMetrics);
+      } else {
+        metrics = {
+          processedCrossChainEvents: [eventID],
+          latency: latency,
+          carbonFootprint: carbonFootprint,
+          cost: cost,
+          throughput: Number(
+            (latency != 0 ? 1 / latency : 0).toFixed(3) as unknown as number,
+          ),
+          latestUpdate: lastAggregated,
+        };
+        this.crossChainModel.setCCTxs(ccTxID, metrics);
+      }
+    });
+    this.crossChainModel.setLastAggregationDate(newAggregationDate);
+    const finalTime = new Date();
+    this.log.debug(
+      `${this.className}-AGGREGATE-CCTX-SUCCESS:${finalTime.getTime() - startTime.getTime()}`,
+    );
+    return;
+  }
+
+  public async persistCrossChainLogCsv(name?: string): Promise<string> {
+    const startTime = new Date();
+    const columns =
+      this.crossChainLog.getCrossChainLogAttributes() as (keyof CrossChainEvent)[];
+    const logName = name
+      ? `${name}.csv`
+      : `hephaestus_log_${startTime.getTime()}.csv`;
+    const csvFolder = path.join(__dirname, "../", "../", "test", "csv");
+    const logPath = path.join(csvFolder, logName);
+    const fnTag = `${this.className}#persistCrossChainLogCsv()`;
+    const ccEvents = this.crossChainLog.logEntries;
+
+    try {
+      // Convert log entries to CSV rows
+      const csvRows = ccEvents.map((entry) => {
+        return columns
+          .map((header) => {
+            const value = entry[header];
+            return typeof value === "string" && value.includes(";")
+              ? `"${value}"`
+              : value;
+          })
+          .join(";");
+      });
+
+      // Concatenate columns and rows into a single CSV string
+      const data = [columns.join(";"), ...csvRows].join("\n");
+      this.log.debug(data);
+
+      // Create directory if it doesn't exist
+      if (!fs.existsSync(csvFolder)) {
+        fs.mkdirSync(csvFolder);
+      }
+      fs.writeFileSync(logPath, data);
+
+      const finalTime = new Date();
+      this.log.debug(
+        `EVAL-${this.className}-PERSIST-LOG-CVS:${finalTime.getTime() - startTime.getTime()}`,
+      );
+      return logName;
+    } catch (error) {
+      const errorMessage = `${fnTag} Failed to export cross-chain event log to CSV file:`;
+      throw new RuntimeError(errorMessage, error);
+    }
+  }
+
+  public async persistCrossChainLogJson(name?: string): Promise<string> {
+    const startTime = new Date();
+    const logName = name
+      ? `${name}.json`
+      : `hephaestus_log_${startTime.getTime()}.json`;
+    const jsonFolder = path.join(__dirname, "../", "../", "test", "json");
```

Comment:
@RafaelAPB Depending on this will break the code in production when it is under a different sub-directory path (./dist/lib/.../) I recommend making it configurable through constructor parameters so that this issue can be avoided